factory/README.md

```diff
@@ -0,0 +1,172 @@
+
+### In the dockerfile
+
+Args can be defined directly in the dockerfile.
+
+dockerfile
+
+```dockerfile
+# Args are defined in the dockerfile before the FROM command.
+# Using these args will cause an image to be created with node, chrome, firefox and edge.
+ARG CHROME_VERSION='107.0.5304.121-1'
+ARG EDGE_VERSION='100.0.1185.29-1'
+ARG FIREFOX_VERSION='107.0'
+
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . -t test
+docker run -it test npm run test -b chrome
+```
+
+### At build time
+
+Args can be passed to the docker build command.
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . --build-arg CHROME_VERSION='107.0.5304.121-1' --build-arg EDGE_VERSION='100.0.1185.29-1' --build-arg FIREFOX_VERSION='107.0' -t test
+
+docker run test npm run test -b chrome
+```
+
+### In docker-compose.yml
+
+Finally, args can be specified in the docker-compose.yml file.
+
+docker-compose.yml
+
+```yml
+version: '3'
+
+services:
+  test:
+    args:
+      CHROME_VERSION: '107.0.5304.121-1'
+      EDGE_VERSION: '100.0.1185.29-1'
+      FIREFOX_VERSION: '107.0'
+    build:
+      context: .
+    command: npm run test
+```
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+docker commands
+
+```bash
+docker-compose build test
+
+docker-compose run test
+```
+
+## API
+
+The following args can be set to define what versions the cypress factory docker includes in it's final build.
+
+If no args are defined, only the default version of node will be installed. This can still be a useful container though since we will also install any required cypress apt dependencies.
+
+### NODE_VERSION
+
+The version of Node to install in the docker container. If not specified, the default version of Node (defined [here](./docker-compose.yml)) is installed. Node is required.
+
+Example: `NODE_VERSION='16.18.1'`
+
+[Node Versions](https://nodejs.org/en/download/releases)
+
+### YARN_VERSION
+
+The version of yarn to install (via npm). If not specified, Yarn is not installed.
+
+Example: `YARN_VERSION='1.22.19'`
+
+[Yarn versions](https://www.npmjs.com/package/yarn)
+
+### CYPRESS_VERSION
+
+The version of Cypress to install (via npm). If not specified, Cypress is not installed.
+
+Example: `CYPRESS_VERSION='12.1.0'`
+
+[Cypress versions](https://www.npmjs.com/package/cypress)
+
+### CHROME_VERSION
```

Comment:
chrome, edge, firefox and yarn don't appear to be installed via npm. copy-paste err?

Reply:
yes